src/directives/BetaOnboard.tsx

```diff
@@ -103,108 +98,127 @@ const BetaOnboard = ({ directive, mutate }: DirectiveProps) => {
                         void mutate();
                     },
                     async (payload: any) => {
+                        const tenantAlreadyTaken = Boolean(
+                            payload?.job_status?.error?.includes(nameTaken)
+                        );
+
+                        // Handle tracking right away
+                        fireGtmEvent('RegisterFailed', {
+                            tenantAlreadyTaken,
+                            tenant: requestedTenant,
+                        });
                         trackEvent(`${directiveName}:Error`, directive);
+
+                        // Update local state
                         setSaving(false);
-                        setServerError(payload.job_status.error);
+                        setServerError(payload?.job_status?.error);
+                        setNameAlreadyTaken(tenantAlreadyTaken);
                     }
                 );
             }
         },
     };
 
-    const nameAlreadyUsed = useMemo(
-        () => serverError?.includes(nameTaken),
-        [serverError]
-    );
```

Comment:
Since we now pass this boolean to gtm I decided to just use state and figure it out inside the callback up above.